python/ray/train/examples/jax_mnist_example.py

```diff
@@ -0,0 +1,220 @@
+# This example showcases how to use Jax (pmap) with Ray Train.
+# Original code: (without pmap)
+# https://github.com/google/flax/blob/main/examples/mnist/train.py
+import argparse
+from typing import Dict
+from ray.air import session
+
+import functools
+import time
+
+from ray.air.config import ScalingConfig
+from ray.train.jax import JaxTrainer
+import einops
+import numpy as np
+import optax
+import tensorflow_datasets as tfds
+
+
+def get_datasets():
+    import jax
+
+    """Load MNIST train and test datasets into memory."""
+    # shard the dataset
+    def shard_fn(x):
+        # shard the dataset for each node (each process)
+        return einops.rearrange(x, "(d l) ... -> d l ...", d=jax.process_count())[
+            jax.process_index()
+        ]
+
+    # Hide any GPUs from TensorFlow. Otherwise TF might reserve memory and make
+    # it unavailable to JAX.
+    import tensorflow as tf
+
+    tf.config.experimental.set_visible_devices([], "GPU")
+
+    ds_builder = tfds.builder("mnist")
+    ds_builder.download_and_prepare()
+    test_ds = tfds.as_numpy(ds_builder.as_dataset(split="test", batch_size=-1))
+    train_ds = tfds.as_numpy(ds_builder.as_dataset(split="train", batch_size=-1))
+
+    # TODO: warning about the sharding dimension
```

Comment:
I think it's okay to add this here and comment that "drop last" semantics are occurring.

Reply:
make sense